middleware/kubernetes/README.md

```diff
@@ -1,204 +1,250 @@
+	kubernetes cluster.local. {
+		endpoint http://k8s-endpoint:8080
+	}
 
-```
-# kubernetes <zone> [<zone>] ...
-#
-# Use kubernetes middleware for domain "coredns.local"
-# Reverse domain zones can be defined here (e.g. 0.0.10.in-addr.arpa),
-# or instead with the "cidrs" option.
-#
-kubernetes coredns.local {
-
-	# resyncperiod <period>
-	#
-	# Kubernetes data API resync period. Default is 5m
-	# Example values: 60s, 5m, 1h
-	#
-	resyncperiod 5m
-
-	# endpoint <url>
-	#
-	# Use url for a remote k8s API endpoint.  If omitted, it will connect to
-	# k8s in-cluster using the cluster service account.
-	#
-	endpoint https://k8s-endpoint:8080
-
-	# tls <cert-filename> <key-filename> <cacert-filename>
-	#
-	# The tls cert, key and the CA cert filenanames for remote k8s connection.
-	# This option is ignored if connecting in-cluster (i.e. endpoint is not
-	# specified).
-	#
-	tls cert key cacert
-
-	# namespaces <namespace> [<namespace>] ...
-	#
-	# Only expose the k8s namespaces listed.  If this option is omitted
-	# all namespaces are exposed
-	#
-	namespaces demo
-
-	# lables <expression> [,<expression>] ...
-	#
-	# Only expose the records for kubernetes objects
-	# that match this label selector. The label
-	# selector syntax is described in the kubernetes
-	# API documentation: http://kubernetes.io/docs/user-guide/labels/
-	# Example selector below only exposes objects tagged as
-	# "application=nginx" in the staging or qa environments.
-	#
-	labels environment in (staging, qa),application=nginx
-
-	# pods <disabled|insecure|verified>
-	#
-	# Set the mode of responding to pod A record requests.
-	# e.g 1-2-3-4.ns.pod.zone.  This option is provided to allow use of
-	# SSL certs when connecting directly to pods.
-	# Valid values: disabled, verified, insecure
-	#  disabled: Do not process pod requests, always returning NXDOMAIN
-	#  insecure: Always return an A record with IP from request (without
-	#            checking k8s).  This option is is vulnerable to abuse if
-	#            used maliciously in conjuction with wildcard SSL certs.
-	#  verified: Return an A record if there exists a pod in same
-	#            namespace with matching IP.  This option requires
-	#            substantially more memory than in insecure mode, since it
-	#            will maintain a watch on all pods.
-	# Default value is "disabled".
-	#
-	pods disabled
-
-	# cidrs <cidr> [<cidr>] ...
-	#
-	# Expose cidr ranges to reverse lookups.  Include any number of space
-	# delimited cidrs, and or multiple cidrs options on separate lines.
-	# kubernetes middleware will respond to PTR requests for ip addresses
-	# that fall within these ranges.
-	#
-	cidrs 10.0.0.0/24 10.0.10.0/25
-
-	# upstream <address> [<address>] ...
-	#
-	# Defines upstream resolvers used for resolving services that point to
-	# external hosts (External Services).  <address> can be an ip, an ip:port, or
-	# a path to a file structured like resolv.conf.
-	upstream 12.34.56.78:53
+### tls
```

Comment:
Bullets are ok for brief explanations for simple options.  But many of the k8s options are quite involved, and require lengthy explanations.  I tried fitting that into bullets, but it was making it very hard to read, so I went with separate sections.

I have included an example section with several full examples.